regression/rt_gk_sheath_1x2v_p1.c

```diff
@@ -202,47 +203,87 @@ create_ctx(void)
 }
 
 void
-evalSourceDensityInit(double t, const double* GKYL_RESTRICT xn, double* GKYL_RESTRICT fout, void* ctx)
+evalElcDensityInit(double t, const double* GKYL_RESTRICT xn, double* GKYL_RESTRICT fout, void* ctx)
 {
   struct sheath_ctx *app = ctx;
   double z = xn[0];
 
-  double n_src = app->n_src;
+  double n_peak = app->n_peak;
   double Lz = app->Lz;
 
   double n = 0.0;
 
-  if (fabs(z) < 0.25 * Lz) {
-    n = n_src;
+  if (fabs(z) <= 0.25 * Lz) {
+    n = 0.5 * n_peak * (1.0 + sqrt(1.0 - (z / (0.25 * Lz)) * (z / (0.25 * Lz)))); // Electron total number density (left).
```

Comment:
the "total" and "left"/"right" in this comment and the one below doesn't make sense to me. I'd remove them.

Reply:
The "left" and "right" descriptors can certainly be removed (I think they are vestigial remnants from some other test, apologies). The "total" descriptor is a convention that I've attempted to enforce throughout the code in order to distinguish between quantities that are being assigned componentwise vs. in one shot. For instance, "Set total drift velocity" means setting all components of the drift velocity simultaneously, while "Set drift velocity (x-direction)" means only setting the x-component. There are quantities, like distribution function values, that are always single-valued in the GK case (and so will always be listed as "total", which may appear redundant), but which are not always single-valued in other cases (such as in PKPM, where the F0 and G functions are set separately). I'd always prefer to err on the side of disambiguation where possible, hence why these comments have been written in this way.
If there is a reason why GK should be regarded as a special case in this respect then I'm happy to hear it; otherwise this is the rationale for the convention.

Reply:
Oh don't even get me started on our use of "vector" for things that are really covectors or pseudovectors. I've given @ammarhakim grief about this before ;)

I thought that it would be useful (as it is in things like 10-moments) to distinguish explicitly between, say, a parallel vs. perpendicular temperature and an isotropic/total temperature. If you say that this distinction is not useful or meaningful in GK, then I can't profess to understand, but I'm not going to contradict you.

Reply:
in GK we simply say the temperature, and I would stick with that. Or the scalar temperature (instead of the tensor temperature).

The left/right above just doesn't make sense to me, sorry. Likewise, so far the distinction between vectors, pseudovectors, covectors, etc, doesn't arise in GK at the input file level, so I would not make the language in the input file more complicated than it needs to be, just because you are using other language with other models. They are different models, with different languages, and more importantly, different communities of people that speak differently (sometimes about the same thing for the bane of us all)

Reply:
Already commented that the left/right thing was a vestigial remnant.

Just because you don't discuss these distinctions doesn't mean that they don't exist, or are not important. There are many places, for instance on the GK side, where a variable will be called something like `T_elc`, and it's not clear whether it's an isotropic temperature that should appear in (e.g.) the calculation of a thermal velocity, or a parallel temperature that should appear (e.g.) in the calculation of a velocity space extent. Very often it's used as both, which is clearly completely wrong.

We should be aiming to minimize these kinds of glaring conceptual errors, not promote them.